core/arch/arm/plat-sam/matrix.c

```diff
@@ -669,8 +224,15 @@ static void matrix_save_regs(vaddr_t base, struct matrix_state *state)
 
 static void matrix_suspend(void)
 {
-	matrix_save_regs(matrix32_base(), &matrix32_state);
-	matrix_save_regs(matrix64_base(), &matrix64_state);
+	unsigned int i = 0;
+	struct matrix *pmatrix = NULL;
+
+	do {
```

Comment:
How about:
``` C
for (pmatrix = matrix_get(i++); pmatrix; pmatrix = matrix_get(i++))
     matrix_save_regs(matrix_base(pmatrix->matrix), &pmatrix->state);
```
same below in `matrix_resume()`.

Reply:
Updated, thank you.